### ISSUE TITLE
Indirection use of `strconv.Atoi` triggers "Incorrect conversion between integer types"

### ISSUE THREAD

**Issue by akram** [argoproj/argo-cd#13136]
* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

Indirection use of `strconv.Atoi` triggers "Incorrect conversion between integer types" by CodeQL/CodeScanning Action.
See : https://github.com/argoproj/argo-cd/pull/13018#discussion_r1157279403

**To Reproduce**

Commit code having use of `strconv.Atoi` like the one present in `env.go`
the CodeQL action will raise an error/warning

**Expected behavior**

The action should run succesfully

**Screenshots**

<img width="744" alt="image" src="https://user-images.githubusercontent.com/654449/230269278-5c2def3a-5734-4ada-96db-537222da2328.png">

**Version**

```shell
argocd: v2.6.7+5bcd846
  BuildDate: 2023-03-23T15:24:49Z
  GitCommit: 5bcd846fa16e4b19d8f477de7da50ec0aef320e5
  GitTreeState: clean
  GoVersion: go1.18.10
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.6.7+5bcd846
  BuildDate: 2023-03-23T15:24:49Z
  GitCommit: 5bcd846fa16e4b19d8f477de7da50ec0aef320e5
  GitTreeState: clean
  GoVersion: go1.18.10
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: could not get kustomize version: exec: "kustomize": executable file not found in $PATH
  Helm Version: could not get helm version: exec: "helm": executable file not found in $PATH
  Kubectl Version: v0.24.2
  Jsonnet Version: v0.19.1```

**Logs**
````
 Check failure

Code scanning/ CodeQL

Incorrect conversion between integer types
Incorrect conversion of an integer with architecture-dependent bit size from [strconv.Atoi](1) to a lower bit size type uint32 without an upper bound check.
```

**Comment by akram** [comment]
This can be fixed by a change in env.go to avoid using strconv.Atoi :

```
func ParseNumFromEnv(env string, defaultValue, min, max int) int {
	str := os.Getenv(env)
	if str == "" {
		return defaultValue
	}
	num, err := strconv.ParseInt(str, 10, 32)
	if err != nil {
		log.Warnf("Could not parse '%s' as a number from environment %s", str, env)
		return defaultValue
	}
	if int(num) < min {
		log.Warnf("Value in %s is %d, which is less than minimum %d allowed", env, num, min)
		return defaultValue
	}
	if int(num) > max {
		log.Warnf("Value in %s is %d, which is greater than maximum %d allowed", env, num, max)
		return defaultValue
	}
	return int(num)
}
```

By using something like this, it fixes the issue, but unit tests are failing then because of max values of int versus max value of int64


It can be a  `good-first-issue` for someone wanting to step into the project

**Comment by Kiruthikameena** [comment]
@akram @crenshaw-dev I am new to the community and looking to contributing. Can I pick this issue?

**Comment by akram** [comment]
hi @Kiruthikameena I will defer the final answer to core contributors. In the meanwhile, I think it is always worth having a look at it and maybe discuss any wider impacts that I have not seen. wdyt?

**Comment by Kiruthikameena** [comment]
@akram just saw that this issue is mentioned in [PR](https://github.com/argoproj/argo-cd/pull/13018). Is this already fixed?

**Comment by akram** [comment]
yes @Kiruthikameena , I had to solve it to make PR #13018 pass all the tests.

**Comment by akshat-g-07** [comment]
Hey @akram ,
Is this issue still open? As you commented earlier that you solved it but I can still see this open.

If it is open, I would like to help.

**Comment by blakepettersson** [comment]
Closing since this issue has been resolved, feel free to reopen if that's not the case.
